src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -247,6 +248,16 @@ bool SolverBase::execute( real64 const time_n,
                           DomainPartition & domain )
 {
   GEOS_MARK_FUNCTION;
+
+  /*
+   * Reset counter indicating the number of cycles since the last timestep cut
+   * when the new timestep. "-1" means that no time-step cut has ocurred.
+   * */
+  if( dt < m_nextDt )
+  {
+    m_lastDtCut = -1;
```

Comment:
so the logic only works with current "execute", no transfer to next cycle?

Reply:
I did this because the current execute might decrease the time-step to meet a exact time target for the event. Then, suppose you come from a recent time-step cut, If I didn't reset `m_lastDtCut`, the code would not be able to increase the time step again (after having decreased not because of a noncoverged Newton, but because it had to meet a target time). Hope this explanation is not too confusing

Reply:
it's ok, just might be confusing that logic is not applied across cycles
let's say we set `minTimeStepIncreaseInterval=5` and it does a cut inside one cycle, then does let's say 2 succesful subtimesteps and goes into another cycle - then that 5-2 counter is lost?

Reply:
Right, in this scenario, the cut information is lost for the next subEvent and I agree it shouldn't. 

Maybe it's possible to improve the logic here. If you have some ideas, feel free to implement!

Reply:
PS: in your scenario, if the time-step is not decreased to meet a exact target time, then the cut information is not lost in the next subEvent